parsec/backend/asgi/error.py

```diff
@@ -0,0 +1,54 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 (eventually AGPL-3.0) 2016-present Scille SAS
+
+from parsec.api.protocol.types import DeviceID, OrganizationID, UserID
+
+
+class HttpProtocolError(Exception):
+    pass
+
+
+class MissingHeaderError(HttpProtocolError):
+    def __init__(self, header: str) -> None:
+        self.header = header
+
+    def __str__(self) -> str:
+        return f"Missing header {self.header}"
+
+
+class InvalidHeaderValue(HttpProtocolError):
+    def __init__(self, header: str, invalid_value: str) -> None:
+        self.header = header
+        self.invalid_value = invalid_value
+
+    def __str__(self) -> str:
+        return f"Invalid header value for header {self.header}"
+
+
+class AuthenticationProtocolError(Exception):
+    pass
+
+
+class DeviceNotFound(AuthenticationProtocolError):
+    def __init__(self, organization_id: OrganizationID, device_id: DeviceID) -> None:
+        self.organization_id = organization_id
+        self.device_id = device_id
+
+    def __str__(self) -> str:
+        return f"Could not find device {self.device_id} in organization {self.organization_id}"
+
+
+class UserIsRevoked(AuthenticationProtocolError):
+    def __init__(self, organization_id: OrganizationID, user_id: UserID) -> None:
+        self.organization_id = organization_id
+        self.user_id = user_id
+
+    def __str__(self) -> str:
+        return f"User {self.user_id} from {self.organization_id} is revoked"
+
+
+class CannotVerifySignature(AuthenticationProtocolError):
+    def __init__(self, internal_exception: Exception) -> None:
+        self.internal_exception = internal_exception
+
+    def __str__(self) -> str:
+        return f"Could not verify the signature, reason: {self.internal_exception}"
```

Comment:
Nice errors :+1: 
Maybe add an inheritance tree in the module docstring:
```
HttpProtocolError
 + MissingHeaderError
 + InvalidHeaderValue
 + AuthenticationProtocolError
   + DeviceNotFound
   + ...
```
Also some errors have the `Error` suffix and some don't. Maybe have `Error` everywhere?

Reply:
I've added the suffix `Error` to those exceptions

Reply:
Adding the _inheritance_ tree in the doc-string would be an extra cost to maintain, I prefer not to.
Modern IDE allow to show references of a item (on `vscode` => `peek references`)

Reply:
> Modern IDE allow to show references of a item

Except it's not a clear at first glance how the exceptions are structured. We have one those tree for pki exceptions already:
https://github.com/Scille/parsec-cloud/blob/890490c21d801fdc2090ec2a4ce4d3ecd159992e/parsec/core/pki/exceptions.py#L3-L41

But I won't fight for it either, it's up to you.